packages/analytics/amplify_analytics_pinpoint_dart/lib/src/impl/analytics_client/event_client/event_storage_adapter.dart

```diff
@@ -64,42 +56,43 @@ class EventStorageAdapter implements Closeable {
   Future<List<StoredEvent>> retrieveEvents({
     int maxEventCount = _maxEventsInBatch,
   }) async {
-    // Raw objects read from Drift storage
-    final driftJsonStrings = await _db.getJsonStrings(maxEventCount);
+    // Raw objects read from storage
+    final queuedItems = await _db.get(maxEventCount);
 
     // Convert raw objects to Event
     final storedEvents = <StoredEvent>[];
-    for (final driftJsonString in driftJsonStrings) {
-      final storedEvent = StoredEvent(driftJsonString, _serializers);
+    for (final item in queuedItems) {
+      final storedEvent = StoredEvent(item, _serializers);
       storedEvents.add(storedEvent);
     }
     return storedEvents;
   }
 
-  /// Delete Events by their DriftId in device storage
-  Future<void> deleteEvents(Iterable<int> idsToDelete) async {
-    await _db.deleteJsonStrings(idsToDelete);
+  /// Delete Events from device storage
+  Future<void> deleteEvents(Iterable<StoredEvent> items) async {
+    final queuedItems = items.map((item) => item.data);
+    return _db.deleteItems(queuedItems);
   }
 
   @override
   Future<void> close() async {
-    await _db.close();
+    return _db.close();
   }
 }
 
 /// Wrapper class around [Event]
 /// Includes DriftId to identify that event for deletion when calling deleteEvents()
 class StoredEvent {
-  /// Create StoredEvent from [DriftJsonString]
-  factory StoredEvent(DriftJsonString data, Serializers serializers) {
-    final event = serializers.deserialize(jsonDecode(data.jsonString)) as Event;
-    return StoredEvent._(data.id, event);
+  /// Create StoredEvent from [QueuedItem]
+  factory StoredEvent(QueuedItem data, Serializers serializers) {
```

Comment:
Because there is extra logic to convert `Event` to `String` and `String` to `Event`. 

Sure it's technically not needed, but I think by OOP it's better to keep that logic separate instead of adding extra complexity to the `EventClient`.